### PR TITLE
added recipe for SavedValues and fixed bug for ManifoldProjection if autodiff==false

### PR DESCRIPTION
**Pull request by ranocha**

### CODE REVIEW

**Comment by ranocha** [comment]
AppVeyor failed due to an `InexactError` while using Plots, see https://github.com/JuliaPlots/Plots.jl/issues/968, https://github.com/JuliaPlots/Plots.jl/issues/963. It is fixed for julia 0.7 in https://github.com/JuliaLang/julia/pull/22644.

File: test/saving_tests.jl [hunk 1]
```diff
@@ -1,11 +1,13 @@
 using Base.Test, OrdinaryDiffEq, DiffEqProblemLibrary, DiffEqCallbacks
+using Plots; unicodeplots()
```
Comment (ChrisRackauckas) [inline test/saving_tests.jl]: ahh, that's a good way to test plotting. I never thought of that before.

File: src/saving.jl [hunk 2]
```diff
@@ -25,6 +25,10 @@ function Base.show(io::IO, saved_values::SavedValues)
                 "\nt:\n", saved_values.t, "\nsaveval:\n", saved_values.saveval)
 end
 
+@recipe function plot(saved_values::SavedValues)
+    (saved_values.t, saved_values.saveval)
```
Comment (ChrisRackauckas) [inline src/saving.jl]: Plots.jl won't plot a `Vector{Vector}` correctly like this. It'll plot each `t` with each `saveval[i]`. If you just make this return `DiffEqArray(saved_values.t, saved_values.saveval)` then it'll do the proper transform for Plots.jl, which requires a transpose and thus requires the indexing. See https://github.com/JuliaDiffEq/RecursiveArrayTools.jl/blob/master/src/vector_of_array.jl#L92.

**Comment by ChrisRackauckas** [comment]
How come the plot isn't showing in the CI output?

**Comment by ranocha** [comment]
I think plots aren't shown in a script, only if `display` (or something similar) is called explicitly.

**Comment by ranocha** [comment]
However, I've tested it in the REPL, the output seems to be okay. I just didn't want to clutter the output if the tests.

**Pull request merged**